BungeeCord-Patches/0017-Micro-optimizations.patch

```diff
@@ -1,13 +1,98 @@
-From 81c109310e7da6e5ab30701bed7a68ec409cc504 Mon Sep 17 00:00:00 2001
+From ba3f7e6ff5f3b9098b35b9b2ac87f9fb2eb22fc1 Mon Sep 17 00:00:00 2001
 From: Tux <write@imaginarycode.com>
 Date: Thu, 19 May 2016 18:05:33 -0600
 Subject: [PATCH] Micro-optimizations
 
 - PluginManager.dispatchCommand() avoids regex while splitting commands. Java 7 introduced an optimized String.split() that should be used instead (affects command dispatch).
 - Avoid regex in getLocale() by changing from replaceAll(String, String) to replaceAll(char, char)
+- Don't attempt to format arguments when there are none provided
+- Don't create a data input stream for every plugin message we get from servers
+- Optimize getUUID() (affects < 1.7.6 clients and IP forwarding)
 
+diff --git a/api/src/main/java/net/md_5/bungee/api/ServerPing.java b/api/src/main/java/net/md_5/bungee/api/ServerPing.java
+index 27b51849..d2a3bc4f 100644
+--- a/api/src/main/java/net/md_5/bungee/api/ServerPing.java
++++ b/api/src/main/java/net/md_5/bungee/api/ServerPing.java
+@@ -5,7 +5,6 @@ import java.util.List;
+ import java.util.UUID;
+ import lombok.AllArgsConstructor;
+ import lombok.Data;
+-import lombok.Getter;
+ import lombok.NoArgsConstructor;
+ import lombok.ToString;
+ import net.md_5.bungee.Util;
+@@ -46,36 +45,54 @@ public class ServerPing
+     }
+ 
+     @Data
+-    @AllArgsConstructor
+     public static class PlayerInfo
+     {
+-
+-        private String name;
+-        private UUID uniqueId;
++        
++        private final String name;
++        private String uniqueId;
++        private String dashedUuid;
++        private boolean undash;
+ 
+         private static final UUID md5UUID = Util.getUUID( "af74a02d19cb445bb07f6866a861f783" );
++        private static final String md5ID = md5UUID.toString();
+ 
+         public PlayerInfo(String name, String id)
+         {
+-            setName( name );
+-            setId( id );
++            this.name = name;
++            this.uniqueId = id;
++            if ( id.indexOf( '-' ) != -1 )
++            {
++                this.undash = true;
```

Comment:
Instead of lazily un-dashing the uuid when getId is finally called, why not do it directly in the constructor?
Also, keep the logic for setting the uuid in the `setId` method, in case we ever need that.